### PR TITLE
Automate rerun of test for stubs update in `Makefile`

### PR DESCRIPTION
**Pull request by sydney-runkle**
Closes https://github.com/pydantic/logfire/issues/469

Also, updated the stubs :)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -36,6 +36,8 @@ test:
 generate-stubs:
 	uv run stubgen -p logfire --include-docstrings --no-analysis
 	rsync -a out/logfire/ logfire-api/logfire_api/
+	uv run pytest ./tests/test_logfire_api.py::test_override_init_pyi || true
```
Comment (sydney-runkle) [inline Makefile:41]: `|| true` so that we ignore the test failure on the first pass, it should report as skipped on the second

File: Makefile [hunk 2]
```diff
@@ -36,6 +36,8 @@ test:
...
+	uv run pytest ./tests/test_logfire_api.py::test_override_init_pyi
```
Comment (sydney-runkle) [inline Makefile:42]: Lmk if you want me to check for `"skipped"` or something here

File: Makefile (hunk 1 above)
Comment (alexmojaki) [reply Makefile:41]: this should be a comment in the code

File: Makefile [hunk 3]
```diff
@@ -36,6 +36,8 @@ test:
 generate-stubs:
 	uv run stubgen -p logfire --include-docstrings --no-analysis
 	rsync -a out/logfire/ logfire-api/logfire_api/
```
Comment (alexmojaki) [inline Makefile:38]:
```suggestion
	rsync -a out/logfire/ logfire-api/logfire_api/
	rm -rf out
```

Sometimes things in `out` from other branches stay there and have caused me confusion.

**Comment by alexmojaki** [comment]
Thanks!

**Pull request merged**